### ISSUE TITLE
topology: add bad configurations to test IPC errors

### ISSUE THREAD

**Issue by plbossart** [thesofproject/sof#682]
add e.g. unsupported MCLKs rates, invalid SSPs or obvious mistakes to check how Linux deals with topology errors. Those bad configurations shall be used only to test that that the driver releases all allocated resources and the modules can be reloaded once a valid topology file is provided.

**Comment by lbetlej** [comment]
to be covered by IPC fuzzer

**Comment by jajanusz** [comment]
@plbossart @lgirdwood @lbetlej Do we need this? Fuzzer should test it without need for keeping bad topologies in repo. Probably we should close it and stick to #1279

**Comment by lgirdwood** [comment]
@jajanusz yep, I think @ranj063 should be able to confirm that fuzzer can take existing valid topollogies and mangle them to check Linux driver error paths.

**Comment by plbossart** [comment]
@jajanusz This is a different level of validation. A bad config can be applied and tested on a regular basis by QA.  It's like module load/unload, it should be part of the basis acceptance tests. 
Fuzzing has elements of randomness to it that will prevent it from being used on a regular basis, and the interpretation of the results could take time.
Different tools really.

**Comment by jajanusz** [comment]
@plbossart You are right, for regular (CI) tests you need something else than fuzzer. I just didn't know what you want to achieve, but now it's clear.

**Comment by cujomalainey** [comment]
As much as I agree with the consistency aspect of targeted tests, we do have a fuzzer running continuously against the FW. I could talk to my security team and see if we can get one for the kernel as well.
